src/Transports/MassTransit.RabbitMqTransport/RabbitMqTransport/RabbitMqBasicConsumer.cs

```diff
@@ -150,15 +137,9 @@ public void HandleBasicCancel(string consumerTag)
 
             CancelPendingConsumers();
 
-            ConsumerCancelled?.Invoke(this, new ConsumerEventArgs(new[] {consumerTag}));
+            ConsumerCancelled?.Invoke(this, new ConsumerEventArgs(new[] { consumerTag }));
 
-            if (_dispatcher.ActiveDispatchCount == 0)
-            {
-                _deliveryComplete.TrySetResult(true);
-                SetCompleted(Task.CompletedTask);
-            }
-            else
-                SetCompleted(_deliveryComplete.Task);
+            TrySetConsumeCanceled();
```

Comment:
This, however, IS the case where the broker canceled us. And in this case, we should be canceling all pending receive context cancellationTokens. (which is what CancelPendingConsumers does, so that's handled above).